### ISSUE TITLE
the `dim` argument in LayerNorm is not used

### ISSUE THREAD

**Issue by csukuangfj** [espnet/espnet#3183]
See
https://github.com/espnet/espnet/blob/55160fc20c48686808cbdf208919e668e3b1451a/espnet/nets/pytorch_backend/transformer/layer_norm.py#L36-L38

It always uses `x.tranpose(1, -1)` instead of `x.transpose(self.dim, -1)`.

It has not caused problems so far since the use case of LayerNorm in ESPnet happens to always be `dim==1`. See
https://github.com/espnet/espnet/blob/55160fc20c48686808cbdf208919e668e3b1451a/espnet/nets/pytorch_backend/fastspeech/duration_predictor.py#L62

https://github.com/espnet/espnet/blob/55160fc20c48686808cbdf208919e668e3b1451a/espnet2/tts/variance_predictor.py#L61

---

I would propose to replace
https://github.com/espnet/espnet/blob/55160fc20c48686808cbdf208919e668e3b1451a/espnet/nets/pytorch_backend/transformer/layer_norm.py#L38

with

```python
        return super(LayerNorm, self).forward(x.transpose(self.dim, -1)).transpose(self.dim, -1)
```

**Comment by sw005320** [comment]
I see.
Thanks for your suggestion.
This sounds better for me.
I'll make a PR.

**Comment by csukuangfj** [comment]
Closed by #3193
